src/uct/ib/dc/dc_mlx5.c

```diff
@@ -177,6 +181,14 @@ uct_dc_mlx5_ep_create_connected(const uct_ep_params_t *params, uct_ep_h* ep_p)
     if_addr    = (const uct_dc_mlx5_iface_addr_t *)params->iface_addr;
     path_index = UCT_EP_PARAMS_GET_PATH_INDEX(params);
 
+    if_addr_max_rd_atomic =
+            if_addr->flags & UCT_DC_MLX5_IFACE_ADDR_MAX_RD_ATOMIC_16 ? 16 : 64;
+    max_rd_atomic = ucs_min(iface->super.super.config.max_rd_atomic,
+                            if_addr_max_rd_atomic);
+
+    uct_dc_mlx5_init_dci_config_key(&dci_config, sl, port_affinity, path_index,
+                                    0, max_rd_atomic);
```

Comment:
maybe pass iface param to uct_dc_mlx5_init_dci_config_key?
to move more "common" logic of initializing the config key inside uct_dc_mlx5_init_dci_config_key

Reply:
I need to delete sl and port affinity from the config key anyway, the other parameters are not derived from iface